### PR TITLE
Initial draft of configuration docs.

### PR DESCRIPTION
**Pull request by jim**
This content in this PR is a summary of [this Slack thread](https://trussworks.slack.com/archives/CDJ9JAD26/p1601998134067100).

### CODE REVIEW

File: developing/configuration/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Configuration
+
+Projects should adhere to the following guidelines for managing configuration:
+
+1. Configuration values should be provided to programs via environment variables.
+2. Programs should validate configuration at boot time.
```
Comment (pkarman) [inline developing/configuration/README.md:6]: is it maybe worth clarifying `program boot time` to distinguish from "OS boot time"?

**Review by pkarman — COMMENTED** [review]
one editorial suggestion.

Is it maybe worth scoping the `.envrc` convention to include a caveat "unless the language/framework already has a well-established convention for managing environment variables". This is the first time I've seen `.envrc` and while it's cool for Golang I just don't know if other server-side apps (e.g. Rails) might have other well-known conventions.

**Comment by macrael** [comment]
@pkarman FWIW .envrc is not at all a Golang convention, it's a truss-infra convention. I think in part because it is language agnostic, so you can use it across both services and tools.

File: developing/configuration/README.md (hunk 1 above)
Comment (cblkwell) [reply developing/configuration/README.md:6]: Isn't this "run time" (ie, when the program is run)? Or am I misunderstanding a nuance?

Comment (jim) [reply developing/configuration/README.md:6]: How about "programs should validate configuration when they are run"? I was trying to communicate that it should occur immediately when the program starts and not at some point later on.

File: developing/configuration/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Projects should adhere to the following guidelines for managing configuration:
+
+1. Configuration values should be provided to programs via environment variables.
+2. Programs should validate configuration immediately when they are run.
+3. If config is incomplete or invalid, programs should:
+   1. Exit with a non-zero status
+   2. Print a helpful error message to assist humans in resolving the issue
+4. All environment values should be cataloged in a single location.
+
+   - This should generally be a project's `.envrc` file (values can be left blank as needed), although ecosystems that have a strong convention for this configuration living elsewhere should follow that convention.
```
Comment (rpdelaney) [inline developing/configuration/README.md:12]:
```suggestion
   - This should generally be a project's `[.envrc](../direnv/README.md)` file (values can be left blank as needed), although ecosystems that have a strong convention for this configuration living elsewhere should follow that convention.
```

**Review by rpdelaney — APPROVED** [review]
FYI I think we are no longer requiring peer review to make changes to the eng playbook.

File: developing/configuration/README.md (hunk 2 above)
Comment (rpdelaney) [reply developing/configuration/README.md:12]: I _think_ that will do what I think it will do... :)

**Comment by macrael** [comment]
I love this, it would be nice to see some examples of the level of description you like for environment variables. On my current project I stole having a separate Configuration.md file that has a couple sentences about each variable in it. I really like your suggestion that that documentation live in .envrc, where all the default vaules for development are going to be laid out, and I think an example would help set the tone for just how much explaination each env var deserves.

**Pull request merged**